builtin/providers/aws/data_source_aws_route_table.go

```diff
@@ -0,0 +1,80 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsRouteTable() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsRouteTableRead,
+
+		Schema: map[string]*schema.Schema{
+			"subnet_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
I am not sure if this should be optional? It seem that rest of the attributes are also optional, is this desirable?